src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -308,6 +315,14 @@ export default class DeployFromSettingsController {
   isPortMappingFilled_(portMapping) { return !!portMapping.port && !!portMapping.targetPort; }
 
   /**
+   * Returns true when the given environment variable is filled by the user, i.e., is not empty.
+   * @param {!backendApi.EnvironmentVariable} variable
+   * @return {boolean}
+   * @private
+   */
+  isVariableFilled_(variable) { return !!variable.name; }
```

Comment:
I use !! here to convert string to a boolean. I want to filter out empty strings and nulls. If I didn't use !! then the returned value wouldn't be {boolean}. With !! it is. 

The unknown is whether variable.name is always string. In this case I'm pretty sure it is. I'll double check however. 

In the case of port mappings, you're right. We should fix it. I'll create a bug.